node/pkg/chain/helper/helper.go

```diff
@@ -240,108 +233,83 @@ func (t *ChainHelper) PublicAddressString() (string, error) {
 	return address.Hex(), nil
 }
 
-func (t *ChainHelper) SubmitDirect(ctx context.Context, contractAddress string, functionString string, args ...interface{}) error {
+func (t *ChainHelper) SubmitDelegatedFallbackDirect(ctx context.Context, contractAddress string, functionString string, args ...interface{}) error {
```

Comment:
I think this should work well but there is no fallback logic if delegator fails. Original logic is if delegator fails then we make direct transaction. But the current logic returns if delegator fails. I'm not going to insist but I think it'd be nice to have the fallback feature (if delegator fails after x number of retries then try with direct). But this is going to be tricky because:
 - if delegator fails during `tx` making phase (all x attempts) then we should reuse the same nonce for direct (because the `tx` hasn't been submitted and the nonce value has to be reused)
 - if delegator fails after `tx` has been submitted to the pool then we should retry with new nonce.

Reply:
but if we are not going to have the fallback feature then i think this implementation should work well